### PR TITLE
view distillation

### PR DESCRIPTION
**Pull request by youny626**

### CODE REVIEW

File: view_distillation/design_doc.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+### Find compatible views
+
+`result = vd.find_compatible_views()`
+
+This will return a list of lists, each inner list contains the views that are identical to each other.
+
+### Prune compatible views
+
+`result = vd.prune_compatible_views()`
```
Comment (raulcf) [inline view_distillation/design_doc.md:25]: a better name would suggest what the summarization operation does, e.g., reduce_compatible_views_to_one()

the pruning aspect is implicit in the operation, but this name explains how the pruning is done

File: view_distillation/design_doc.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+This will prune all compatible views and leave one, any future API calls will operate on the rest of views.
+It will also return the list of views left after pruning.
+
+### Find contained views
+
+`result = vd.find_compatible_views()`
```
Comment (raulcf) [inline view_distillation/design_doc.md:32]: typo, this is contained views

File: view_distillation/design_doc.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+This will return a list of lists, each inner list contains the set of views in sorted order (descending by view size)
```
Comment (raulcf) [inline view_distillation/design_doc.md:34]: confusing use of list and set:

- First, why is the first list a list and not a set? 
- Second, the "inner list returns a set of views sorted" does not make sense. If it's sorted it is a list, not a set, right?

File: view_distillation/design_doc.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+This will prune all compatible views and leave one, any future API calls will operate on the rest of views.
+It will also return the list of views left after pruning.
+
+### Find contained views
+
+`result = vd.find_compatible_views()`
+
+This will return a list of lists, each inner list contains the set of views in sorted order (descending by view size) 
+where each view contains all the following views.
```
Comment (raulcf) [inline view_distillation/design_doc.md:35]: what is "following view"

it is important to spend time thinking carefully how we are explaining these APIs to the world

File: view_distillation/design_doc.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+### Prune contained views
+
+`result = vd.prune_contained_views()`
```
Comment (raulcf) [inline view_distillation/design_doc.md:39]: as above, the pruning is not clear. it is better if it is obvious. if there are multiple summarization strategies, those could be given as a parameter (even if we only implement one now)

**Review by raulcf — COMMENTED** [review]
is the summarization strategy a first-class object of this interface? it may need to be, in which case, that should be reflected in the design.

File: view_distillation/code_used_for_paper/utils.py [hunk 6]
```diff
@@ -0,0 +1,399 @@
+import random
```
Comment (raulcf) [inline view_distillation/code_used_for_paper/utils.py:1]: this utils.py is in a folder "code_used_for_paper" -- can you refactor this asap?

File: view_distillation/code_used_for_paper/utils.py [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+
+from collections import namedtuple, defaultdict
+
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+
+def normalize(df):
+    # df = df.astype(str).apply(lambda x: x.str.strip().str.lower())
```
Comment (raulcf) [inline view_distillation/code_used_for_paper/utils.py:12]: clean up commented out code

File: view_distillation/code_used_for_paper/utils.py [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+
+def add_to_row_to_path_dict(row_to_path_dict, row_strs, path):
+    for row in row_strs:
+        if row not in row_to_path_dict.keys():
+            row_to_path_dict[row] = {path}
+        else:
+            row_to_path_dict[row].add(path)
+
+
+def create_contradictory_signals(path_to_df_dict,
```
Comment (raulcf) [inline view_distillation/code_used_for_paper/utils.py:79]: are all these functions used? would be good to verify that is the case

**Review by raulcf — CHANGES_REQUESTED** [review]
All functionality seems concentrated in a utils.py. This is not a very good design. Separate the main logic of the module from the real "utils". I'll take a look at the PR after.

**Comment by youny626** [comment]
> All functionality seems concentrated in a utils.py. This is not a very good design. Separate the main logic of the module from the real "utils". I'll take a look at the PR after.

@raulcf I think you are looking at the wrong file. I am not using utils.py(or anything else) from code_used_for_paper/. The new module only uses utils.py and view_distillation.py under the current view_distillation directory. Documentations are in README.

**Pull request closed without merging**